app/jobs/publish_feed_job.rb

```diff
@@ -19,13 +26,17 @@ def publish_feed(podcast, feed)
   def publish_apple(feed)
     feed.apple_configs.map do |config|
       if feed.publish_to_apple?(config)
-        PublishAppleJob.perform_later(config)
+        res = PublishAppleJob.perform_now(config)
+        PublishingPipelineState.publish_apple!(feed.podcast)
+        res
       end
     end
   end
 
   def publish_rss(podcast, feed)
-    save_file(podcast, feed)
+    res = save_file(podcast, feed)
+    PublishingPipelineState.publish_rss!(podcast)
```

Comment:
Okay, so you could end up with multiple `PublishingPipelineState`s with status `:publishing_rss` ... if your podcast has > 1 feed.  But that sort of makes sense... we're logging that "hey, we wrote these 3 RSS feeds", and then move onto the publish-apple stage.

Reply:
Yep, that's right.  So you'd see a row for each thing in the pipeline. 

I could imagine some structured output here as well -- at the level of the state. Either as a json column on the pipeline state, or an association.  Things like the feed id, how many bytes were written, how many episodes in the RSS, was it private when it was written, etc.